scripts/o3de/o3de/enable_gem.py

```diff
@@ -120,8 +120,8 @@ def enable_gem_in_project(gem_name: str = None,
             # it takes about 150ms to populate this structure with 137 gems, 4696 bytes in total
             all_gems_json_data = manifest.get_gems_json_data_by_name(project_path=project_path, include_manifest_gems=True, include_engine_gems=True)
 
-            # remove any gems that are not active or dependencies
-            manifest.remove_non_dependency_gem_json_data(enabled_gem_names, all_gems_json_data)
+            # we don't remove gems that are not active or dependencies
+            # because they will be implicitely found and activated via cmake 
```

Comment:
nit: This comment may be mis-interpreted to describe the ```incompatible_objects=``` line below it. Suggest making a note: here instead 
```suggestion
            # Note: we don't remove gems that are not active or dependencies
            #       because they will be implicitely found and activated via cmake 
```